cloud/linode/loadbalancers.go

```diff
@@ -274,6 +276,13 @@ func (l *loadbalancers) EnsureLoadBalancerDeleted(ctx context.Context, clusterNa
 	sentry.SetTag(ctx, "cluster_name", clusterName)
 	sentry.SetTag(ctx, "service", service.Name)
 
+	// Don't delete the underlying nodebalancer if the service has the preserve annotation.
+	if preserveRaw, ok := service.Annotations[annLinodeLoadBalancerPreserve]; ok {
+		if preserve, err := strconv.ParseBool(preserveRaw); err == nil && preserve {
+			return nil
+		}
```

Comment:
Missing a `return err` branch here.

Reply:
This was intentional. When talking with @jnschaeffer, we thought it'd be better to interpret an invalid boolean as `false` and document it as such. Do you think there is value in throwing an error here instead?

Reply:
I had this comment as pending for a while. Sorry for the confusion in chat.

Reply:
I added a note below the annotation lists describing the accepted string representations for `true`, [here](https://github.com/linode/linode-cloud-controller-manager/pull/69/files#diff-04c6e90faac2675aa89e2176d2eec7d8R66).